bioindex/lib/index.py

```diff
@@ -210,37 +210,36 @@ def delete_stale_keys(self, engine, objects, console=None):
          - hasn't been fully indexed
         """
         logging.info('Finding stale keys...')
-        keys = self.lookup_keys(engine)
-        # if the file in s3 is not the db, it's new and we need to index
-        # if a file in s3 is in the db but the version is different, we need to index
-        new_updated_or_deleted_files = [o for o in objects if o['Key'] not in keys
-                                        or keys[o['Key']]['version'] != o['ETag'].strip('"')]
-        # find files that are in the db but not in s3
-        deleted_files = set(keys) - set([o['Key'] for o in objects])
-        new_updated_or_deleted_files.extend([{'Key': k} for k in deleted_files])
-
-        # delete stale keys
-        if new_updated_or_deleted_files:
+        db_keys = self.lookup_keys(engine)
+        # if a file in s3 is in the db but the version is different from what's in s3 we delete
+        updated_files = [{'id': db_keys[o['Key']]['id'], 'key': o['Key']} for o in objects
+                         if o['Key'] in db_keys and db_keys[o['Key']]['version'] != o['ETag'].strip('"')]
```

Comment:
Maybe a bit clearer if you have three lines here:
```
updated_files = [{'id': db_keys[o['Key']]['id'], 'key': o['Key']} for o in objects
                         if o['Key'] in db_keys and db_keys[o['Key']]['version'] != o['ETag'].strip('"')]
deleted_files = [{'id': db_keys[k]['id'], 'key': k} for k in db_keys if k not in s3_keys]
new_files = [o for o in objects if o['Key'] not in db_keys]
```
Then define
```
updated_or_new_files = updated_files + new_files
updated_or_deleted_files = updated_files + deleted_files
```
then you return `updated_or_new_files`